### PR TITLE
feat(x/gov): adds constitution to x/gov

### PR DESCRIPTION
**Pull request by adam-hanna**
Please go the the `Preview` tab and select the appropriate sub-template:

* [Production code](?expand=1&template=production.md) - for types `fix`, `feat`, and `refactor`.
* [Docs](?expand=1&template=docs.md) - for documentation changes.
* [Others](?expand=1&template=others.md) - for changes that do not affect production code.

### CODE REVIEW

File: proto/atomone/gov/v1/genesis.proto [hunk 1]
```diff
@@ -30,4 +30,11 @@ message GenesisState {
   //
   // Since: cosmos-sdk 0.47
   Params params = 8;
+  // The constitution allows builders to lay a foundation and define purpose.
+  // This is an immutable string set in genesis.
+  // There are no amendments, to go outside of scope, just fork.
+  // constitution is an immutable string in genesis for a chain builder to lay out their vision, ideas and ideals.
```
Comment (tbruyelle) [inline proto/atomone/gov/v1/genesis.proto:36]: This comment is irrelevant for AtomOne, the string is mutable as we will have amendments.

File: x/gov/types/keys.go [hunk 2]
```diff
@@ -51,6 +51,9 @@ var (
 
 	// ParamsKey is the key to query all gov params
 	ParamsKey = []byte{0x30}
+
+	// KeyConstitution is the key string used to store the chain's constitution
+	KeyConstitution = []byte("constitution")
```
Comment (tbruyelle) [inline x/gov/types/keys.go:56]: Usually keys are numbers, but can't tell you exactlty what is the reason behind (key size?)

**Review by tbruyelle — CHANGES_REQUESTED** [review]
Just some minor changes, but overall your change is good.

Once this PR and PR #11 have been merged, there's a next step: constitutional amendment proposals to update the constitution in the store.

File: x/gov/types/keys.go (hunk 2 above)
Comment (giunatale) [reply x/gov/types/keys.go:56]: Yes, key size

**Comment by adam-hanna** [comment]
> Just some minor changes, but overall your change is good.
> 
> Once this PR and PR #11 have been merged, there's a next step: constitutional amendment proposals to update the constitution in the store.

Thank you!

Ok, changes made :)

**Review by tbruyelle — APPROVED** [review]

**Pull request merged**